### PR TITLE
Enforce building of benchmark programs in CI

### PR DESCRIPTION
**Pull request by erikd**
## Description

Up until now, benchmark programs have not been built in CI. This meant that changes to a library that breaks a benchmark would often go un-noticed for some time.

This PR makes sure all benchmark programs are built, but not run.

## Linked issue

N/A

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [x] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -31,6 +32,13 @@ let
       kill $TAILPID
     '';
   });
+  applyBenchmarks = drv: overrideCabal drv (drv: {
+    configureFlags = (drv.configureFlags or []) ++ [
+      "--enable-benchmarks"
+    ];
+    doBenchmark = true;
```
Comment (avieth) [inline default.nix]: I think this could use a comment. Does `doBenchmark = true` not imply the `--enable-benchmarks` flag?

File: default.nix [hunk 2]
```diff
@@ -89,6 +98,30 @@ let
...
+               # Filter out the files which I'm editing often.
+               lib.hasSuffix ".nix" baseName ||
+               # Filter out nix-build result symlinks
+               (type == "symlink" && lib.hasPrefix "result" baseName)
+             );
+
+          in
+            if (builtins.typeOf args.src) == "path"
+              then builtins.filterSource cleanSourceFilter args.src
+              else args.src or null;
```
Comment (avieth) [inline default.nix]: Is this needed in order to enable benchmarks?

File: networking/cardano-sl-networking.cabal [hunk 3]
```diff
@@ -269,6 +270,8 @@ test-suite cardano-sl-networking-test
                      , random
                      , safe-exceptions
                      , serokell-util >= 0.1.2.3
+                     -- problem with stack2nix benchmarks
```
Comment (avieth) [inline networking/cardano-sl-networking.cabal]: What's the problem?

**Comment by erikd** [comment]
Closing this PR as the last commit proves that benchmark programs are now being build in CI.

**Pull request closed without merging**